packages/cache/src/internal/tar.ts

```diff
@@ -62,6 +62,7 @@ export async function extractTar(
     }
   }
   const args = [
+    '--posix',
```

Comment:
What happens if `--posix` is called on an archive that wasn't created with the `--posix` flag?

Reply:
I will check it.

Reply:
`--posix` is not allowed when extracting
```
>tar --posix -xzf qwe.tgz qwe                   
tar: Option --format is not permitted in mode -x
```

Reply:
Oops, all right. I will update this PR.